source/rust_verify/src/rust_to_vir_base.rs

```diff
@@ -553,73 +427,204 @@ pub(crate) fn is_smt_arith<'tcx>(bctx: &BodyCtxt<'tcx>, id1: &HirId, id2: &HirId
-            let path = &trait_ref.path;
-            let def_id = match path.res {
-                rustc_hir::def::Res::Def(_, def_id) => def_id,
-                _ => return unsupported_err!(span, "generic bounds"),
-            };
-            if Some(def_id) == tcx.lang_items().fn_trait() {
-                let args = &path.segments.last().expect("last segment").args.expect("GenericArgs");
-                unsupported_err_unless!(args.args.len() == 1, span, "generic bounds");
-                unsupported_err_unless!(args.bindings.len() == 1, span, "generic bounds");
-                unsupported_err_unless!(
-                    args.bindings[0].gen_args.args.len() == 0,
-                    span,
-                    "generic bounds"
-                );
-                let t_args = match &args.args[0] {
-                    rustc_hir::GenericArg::Type(t) => ty_to_vir(tcx, &t),
-                    _ => panic!("unexpected arg to Fn"),
-                };
-                let t_ret = match &args.bindings[0].kind {
-                    rustc_hir::TypeBindingKind::Equality { ty } => ty_to_vir(tcx, ty),
-                    _ => panic!("unexpected arg to Fn"),
-                };
-                let args = match &*t_args {
-                    TypX::Tuple(args) => args.clone(),
-                    _ => panic!("unexpected arg to Fn"),
-                };
-                Ok(Arc::new(GenericBoundX::FnSpec(args, t_ret)))
-            } else if Some(def_id) == tcx.lang_items().sized_trait()
-                || Some(def_id) == tcx.lang_items().copy_trait()
-                || Some(def_id) == tcx.lang_items().unpin_trait()
-                || Some(def_id) == tcx.lang_items().sync_trait()
-                || Some(def_id) == tcx.get_diagnostic_item(rustc_span::sym::Send)
-            {
-                // Rust language marker traits are ignored in VIR
-                Ok(Arc::new(GenericBoundX::Traits(vec![])))
-            } else {
-                let typx = def_id_to_datatype_typx(tcx, def_id, &path.segments);
-                if let TypX::Datatype(trait_name, _args) = typx {
-                    Ok(Arc::new(GenericBoundX::Traits(vec![trait_name])))
-                } else {
-                    panic!("unexpected trait bound");
-                }
-            }
-        }
-        _ => {
-            unsupported_err!(span, "generic bounds")
-        }
+    if Some(trait_def_id) == tcx.lang_items().sized_trait()
+        || Some(trait_def_id) == tcx.lang_items().copy_trait()
+        || Some(trait_def_id) == tcx.lang_items().unpin_trait()
+        || Some(trait_def_id) == tcx.lang_items().sync_trait()
+        || Some(trait_def_id) == tcx.get_diagnostic_item(rustc_span::sym::Send)
+    {
+        // Rust language marker traits are ignored in VIR
+        Ok(Arc::new(GenericBoundX::Traits(vec![])))
+    } else {
+        // REVIEW: this seems wrong
+        // Why did the previous code here throw away the args?
+        let _args = args;
```

Comment:
We don't yet include trait arguments in the VIR AST because we only support very simple uses of traits for now.  For example, if you say:
```
        trait T<A> { fn f(&self, x: &A); }
        struct S {}
        impl T<bool> for S { fn f(&self, x: &bool) {} }
        impl T<u8>   for S { fn f(&self, x: &u8)   {} }
```
Verus complains:
```
error: not supported: multiple definitions of same function
   --> .\rust_verify\example\features.rs:115:13
    |
112 |             fn f(&self, x: &bool) {}
    |             ---------------------
...
115 |             fn f(&self, x: &u8) {}
    |             ^^^^^^^^^^^^^^^^^^^
```